airbyte-config/persistence/src/test/resources/postgres_ssh_key/spec.json

```diff
@@ -0,0 +1,220 @@
+      "airbyte_secret": true,
+      "order": 4
+    },
+    "ssl": {
+      "title": "Connect using SSL",
+      "description": "Encrypt client/server communications for increased security.",
+      "type": "boolean",
+      "default": false,
+      "order": 5
+    },
+    "replication_method": {
+      "type": "object",
+      "title": "Replication Method",
+      "description": "Replication method to use for extracting data from the database.",
+      "order": 6,
+      "oneOf": [
+        {
+          "title": "Standard",
+          "additionalProperties": false,
+          "description": "Standard replication requires no setup on the DB side but will not be able to represent deletions incrementally.",
+          "required": ["method"],
+          "properties": {
+            "method": {
+              "type": "string",
+              "const": "Standard",
+              "enum": ["Standard"],
+              "default": "Standard",
+              "order": 0
+            }
+          }
+        },
+        {
+          "title": "Logical Replication (CDC)",
+          "additionalProperties": false,
+          "description": "Logical replication uses the Postgres write-ahead log (WAL) to detect inserts, updates, and deletes. This needs to be configured on the source database itself. Only available on Postgres 10 and above. Read the <a href=\"https://docs.airbyte.io/integrations/sources/postgres\">Postgres Source</a> docs for more information.",
+          "required": ["method", "replication_slot", "publication"],
+          "properties": {
+            "method": {
+              "type": "string",
+              "const": "CDC",
+              "enum": ["CDC"],
+              "default": "CDC",
+              "order": 0
+            },
+            "plugin": {
+              "type": "string",
+              "description": "A logical decoding plug-in installed on the PostgreSQL server. `pgoutput` plug-in is used by default.\nIf replication table contains a lot of big jsonb values it is recommended to use `wal2json` plug-in. For more information about `wal2json` plug-in read <a href=\"https://docs.airbyte.io/integrations/sources/postgres\">Postgres Source</a> docs.",
+              "enum": ["pgoutput", "wal2json"],
+              "default": "pgoutput",
+              "order": 1
+            },
+            "replication_slot": {
+              "type": "string",
+              "description": "A plug-in logical replication slot.",
+              "order": 2
+            },
+            "publication": {
+              "type": "string",
+              "description": "A Postgres publication used for consuming changes.",
+              "order": 3
+            }
+          }
+        }
+      ]
+    },
+    "tunnel_method": {
+      "type": "object",
+      "title": "SSH Tunnel Method",
+      "description": "Whether to initiate an SSH tunnel before connecting to the database, and if so, which kind of authentication to use.",
+      "oneOf": [
+        {
```

Comment:
Here is where the problem was. It was expecting a `"type": "object"` here, which is missing. However, it isn't technically required by JSONSchema, so my implementation was wrong.